query-engine/connector-test-kit/src/test/scala/util/SchemaBaseV11.scala

```diff
@@ -225,11 +227,12 @@ trait SchemaBaseV11 extends PlayJsonExtensions {
                                     childId <- idOptions;
 
                                     // Based on Id and relation fields
-                                    childReferenceToParent  <- childReferences(simple, parentId, onParent, onChild);
+                                    childReferenceToParent <- childReferences(simple, parentId, onParent, onChild);
                                     parentReferenceToChild <- parentReferences(simple, childId, childReferenceToParent, onParent, onChild);
 
-                                    // skip required virtual relation fields as those are not allowed
-                                    if onParent.isRequired && parentReferenceToChild != noRef;
+                                    // skip required virtual relation fields as those are disallowed in a Prisma Schema
+                                    isVirtualRequiredRelationField = onParent.isRequired && parentReferenceToChild == noRef
+                                    if !isVirtualRequiredRelationField;
```

Comment:
This was totally broken imo. It was intended to skip these kinds of schemas:
```prisma
model Parent {
   id    Int @id
   ...
   child Child // <-- this is a virtual required relation field
}
```
But instead the logic rather applied to those schemas if I understand this correctly:
```prisma
model Parent {
   id    Int @id
   ...
   child Child @relation(fields: [...], references: [...]) // <-- exactly not a required virtual relation field :-(
}
```

Reply:
Yeah, that's the way I understand it too.